segment.go

```diff
@@ -581,6 +585,28 @@ func (s *SegmentBase) Count() uint64 {
 	return s.numDocs
 }
 
+func (s *SegmentBase) getMaxDocID(idDict *Dictionary) (string, error) {
+	// max key is cached, return it
+	if s.cachedMaxDocID != "" {
```

Comment:
shouldn't this be under lock (also is this a duplicate of the following lines)?

Reply:
Ya, looks like there could be a data race otherwise.

Reply:
sorry